src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java

```diff
@@ -195,29 +205,33 @@ public void safeUpdate() {
         }
         */
 
-        ItemStack stackInInput = getStackInSlot(1);
-        if (FluidHelper.itemStackIsWaterContainer(stackInInput)) {
-            ItemStack drainedItemStack = FluidHelper.fillTankFromItem(stackInInput, myTank, true);
-            setInventorySlotContents(1, drainedItemStack);
+        ItemStack stackInInput = inventory.getStackInSlot(1);
+        if (!stackInInput.isEmpty()) {
```

Comment:
This line is unnecessary, `FluidHelper.itemStackIsWaterContainer(ItemStack)` checks if the provided `ItemStack` is empty.

Reply:
It would be empty if no item is placed in the slot, no?

Reply:
Right, what I mean is that `FluidHelper.itemStackIsWaterContainer(ItemStack)` returns false if `ItemStack#isEmpty`, so there's no need to check it twice.

Reply:
I guess yeah.